### ISSUE TITLE
v7+HMMA: GP_GET advances but payload doesn't land at op 1 (addrelu)

### ISSUE THREAD

**Issue by johnjezl** [SLM-OS/SLM-Operating-System#710]
## Summary

After PR #694 lands the v7 QMD-pool encoder fix and the v7 acceptance in the scanner-level validator (commit 8ef5f99e), end-to-end v7+HMMA on jetson-nano-2 (`kexec --no-gpu-suspend SLMOS_GEMM_TIER=hmma`) reaches the channel-inherit successfully, but the first user dispatch fails at op 1 (`01_addrelu1`):

```
[GA10B-P6] Found handoff at phys 0x127e06000 (kind=0)
[GA10B-P6] channel handoff valid — inherited from Linux
[GA10B-P8] GP_GET advanced but payload didn't land (got 0x0, want 0xcafedead) — PBDMA consumed our submit but the GPU didn't produce the expected side effect.
[GA10B-P8] pipeline op 1 failed (rc=-1) — aborting chain
[engine] mnist GPU fastpath failed -- falling back to CPU
```

Both warmup, throwaway, and user dispatches fail with the same error. CPU fallback produces sane logits.

## What works

- Manual standalone helper run (v6 mode, helper-baked QMDs) on the same board shows HMMA tensor cores correctly executing: GPU vs CPU logits diverge by ~`0.000410` (ULP-level FP16-weight magnitude), argmax matches both. So the HMMA SASS itself is correct.
- v7 channel handoff is discovered, validated, and inherited cleanly.
- Op 0 (`00_conv1`) appears to dispatch — failure is at op 1.

## What's broken

The v7 per-dispatch QMD pool path produces dispatches whose PBDMA consumes the pushbuffer but the GPU doesn't write the expected payload. Suspected areas:
- QMD slot rotation (`(slot + 1u) % pool_n_slots`) — slot 0 may have been preheated by warmup, slot 1's QMD encoding may be stale or wrong.
- The HMMA overrides apply to the GEMM op (op 6) only — but op 1 fails before HMMA is even reached, suggesting this is an issue with the v7 pool dispatch generally, not HMMA-specific.
- Cache flushing of the per-slot QMD before SEND_PCAS — `slot.cpu_va` is flushed via `gsp_platform->cache_clean(...)`. May need additional barriers.
- Pushbuffer encoding for the per-op slot — each op's pushbuffer is at `pushbuf_phys + i * pb_kernel_bytes`. May overlap or have stale GPU VAs.

## Reproducer

```bash
# On jetson-nano-2
ssh root@192.168.4.5
SLMOS_GEMM_TIER=hmma slmos-kexec --no-gpu-suspend
# At slmos> prompt:
gpu use inference on
lua-admin /mnt/files/all_logits.lua  # or any slm.model_infer call
```

Expected: GPU dispatch succeeds, logits diverge from CPU by ~1e-3.
Actual: All ops fail at op 1, fallback to CPU.

## Related

- PR #694 — v7 HMMA structural fix (lands the encoder + validator changes that get us to this point)
- #692 — original "GPU MNIST returns CPU-identical logits" investigation (now mostly explained: prior runs were in suspend-kexec mode where channel inherit fails outright; this issue blocks the `--no-gpu-suspend` path)
- #702 — `compute_ready` audit (orthogonal — that's about eligibility-gate diagnostics)

## Acceptance criteria

`gpu use inference on` followed by an MNIST inference call returns logits that:
- Differ from CPU by ~1e-3 magnitude (FP16-weight HMMA signature)
- Match argmax
- No `[engine] mnist GPU fastpath failed` log line on serial

**Comment by johnjezl** [comment]
Fixed by PR #694 commit e5384f55. End-to-end verified on jetson-nano-2 with SLMOS_GEMM_TIER=hmma + --no-gpu-suspend:

```
[GA10B-P8-v7] BULK 8 ops + 1 sema — gp_put=35, ...
[GA10B-P8-v7] BULK pipeline complete — 8 ops fired, GP_GET=44
argmax=1
  logit[0] = -0.9124786
  logit[1] = 13.0703344
  ...
```

No `[engine] mnist GPU fastpath failed` log line. Engine returns GPU-derived logits.

Two bugs were the cause:
1. `ga10b_bringup_channel_kind` never copied the v7 tail fields (`qmd_pool_phys`, `qmd_pool_gpu_va`, `qmd_pool_size_bytes`, `qmd_pool_n_slots`) into `g_handoff`. With those zero, `ga10b_handoff_is_v7` returned false → v5/v6 fallback dispatch → empty per-op qmd_gpu_va → 'payload didn't land'.
2. `ga10b_bringup_read_pipeline_output` indexed the v7 ops array (80-byte stride) using v6 stride (24 bytes). `ops[7]` under v6-cast lands inside v7 op[2]'s tail; helper sets v7 `expected_payload=0` per op so the read returned 0 → `output_phys==0` → -1 → engine 'fastpath failed' even though the GPU dispatch actually completed.

A separate input-staleness issue (GPU produces identical logits regardless of `slm_gpu_set_mnist_input` value) was discovered during verification — will file as a follow-up. Out of scope for this issue.
